### PR TITLE
Add RuboCop config for this project, run RuboCop

### PR DESCRIPTION
**Pull request by robhanlon22**
Closes #4.

### CODE REVIEW

File: shared/.rubocop.yml [hunk 1]
```diff
@@ -55,9 +60,6 @@ Style/DateTime:
 Style/IfUnlessModifier:
   Enabled: false
 
-Style/EachWithObject:
```
Comment (robhanlon22) [inline shared/.rubocop.yml:63]: This was duplicated

File: bin/changelog-config [hunk 2]
```diff
@@ -63,7 +66,7 @@ parse_lines(lines) do |line, idx|
   elsif summary?(line)
     curr_release['summary'] = line
   elsif entry?('Fixed', line)
-    parse_lines(lines[idx+1..-1], skip_empty: false) do |rl|
+    parse_lines(lines[idx + 1..-1], skip_empty: false) do |rl|
```
Comment (flash-gordon) [inline bin/changelog-config]:
```suggestion
    parse_lines(lines[(idx + 1)..-1], skip_empty: false) do |rl|
```

File: bin/changelog-config [hunk 3]
```diff
@@ -73,7 +76,7 @@ parse_lines(lines) do |line, idx|
       end
     end
   elsif entry?('Added', line)
-    parse_lines(lines[idx+1..-1], skip_empty: false) do |rl|
+    parse_lines(lines[idx + 1..-1], skip_empty: false) do |rl|
```
Comment (flash-gordon) [inline bin/changelog-config]:
```suggestion
    parse_lines(lines[(idx + 1)..-1], skip_empty: false) do |rl|
```

File: bin/changelog-config [hunk 4]
```diff
@@ -83,7 +86,7 @@ parse_lines(lines) do |line, idx|
       end
     end
   elsif entry?('Changed', line)
-    parse_lines(lines[idx+1..-1], skip_empty: false) do |rl|
+    parse_lines(lines[idx + 1..-1], skip_empty: false) do |rl|
```
Comment (flash-gordon) [inline bin/changelog-config]:
```suggestion
    parse_lines(lines[(idx + 1)..-1], skip_empty: false) do |rl|
```

**Review by flash-gordon — APPROVED** [review]

File: .rubocop [hunk 5]
```diff
@@ -0,0 +1 @@
+--ignore-parent-exclusion
```
Comment (robhanlon22) [inline .rubocop:1]: This forces the root project to not exclude any files so RuboCop can be correctly run on the `shared` directory

**Comment by robhanlon22** [comment]
Thanks for the review, @flash-gordon! I've applied your suggestions and added a few more small commits.

**Pull request merged**